src/inmanta/agent/handler.py

```diff
@@ -145,15 +149,57 @@ def wrapper(self: HandlerAPI[TResource], *args: object, **kwds: object) -> objec
             def bound(**kwds: object) -> object:
                 return f(self, **kwds)
 
+            _evict_after_last_access: float
+            _evict_after_creation: float
+
+            # Legacy `for_version` parameter is used, compute
+            # evict_after_last_access and evict_after_creation
+            if for_version is not None:
+                if for_version:
+                    _evict_after_creation = 0
+
+                    if evict_after_last_access > 0:
+                        _evict_after_last_access = evict_after_last_access
+                    else:
+                        _evict_after_last_access = 60
+                else:
+                    _evict_after_last_access = 0
+
+                    if evict_after_creation > 0:
+                        _evict_after_creation = evict_after_creation
+                    elif timeout and timeout > 0:
+                        _evict_after_creation = timeout
+                    else:
+                        _evict_after_creation = 5000
+
+                    if evict_after_creation > 0 and timeout and timeout > 0:
+                        LOGGER.warning(
+                            msg="Both the `evict_after_creation` and the deprecated `timeout` parameter are set "
+                            "for cached method %s. Cached entries will be kept in the cache for %.2fs "
+                            "after entering it." % (f.__name__, evict_after_creation)
```

Comment:
I know this is logically equivalent to what you wrote, but it's more easy to verify it's correct like this.
```suggestion
                            "after entering it." % (f.__name__, _evict_after_creation)
```